spec-main/api-autoupdater-darwin-spec.ts

```diff
@@ -0,0 +1,263 @@
+import { expect } from 'chai'
+import * as cp from 'child_process'
+import * as http from 'http'
+import * as express from 'express'
+import * as fs from 'fs-extra'
+import * as os from 'os'
+import * as path from 'path'
+import { AddressInfo } from 'net';
+
+const features = process.atomBinding('features')
+
+const fixturesPath = path.resolve(__dirname, '../spec/fixtures')
+
+// We can only test the auto updater on darwin non-component builds
+const describeFn = (process.platform === 'darwin' && !process.mas && !features.isComponentBuild() ? describe : describe.skip)
+
+describeFn('autoUpdater behavior', function () {
+  this.timeout(120000)
+
+  let identity = ''
+
+  beforeEach(function () {
```

Comment:
does this need to be rerun before every test or can we just do it once at the beginning?

Reply:
I did it in a `beforeEach` so that `throw new Error('No valid signing identity available to run autoUpdater specs')` runs on every test and marks them all as failed (6 failed tests) instead of a `before` which would cause (1 failed test) and be less obvious as a legit failure.

The time taken to run that script is a few ms, no noticeable performance impact when the tests themselves like a number of seconds to run each time

Reply:
IMO this is of dubious value and rather seems to me like a bug in mocha that failing a `before` only shows up as a single failure rather than N failures.